Wifi/Main.hs

```diff
@@ -6,12 +6,35 @@ import           Data.Monoid
 import           Data.Text.IO (putStrLn)
 import           Prelude      hiding (putStrLn)
 import           Turtle
+import qualified Data.Text as Text
+import           Data.Maybe
+import           System.Environment (getArgs)
 
 main :: IO ()
-main = sh $ liftIO . putStrLn . format' =<< ssid
+main = do
+  args <- getArgs
+  sh $ liftIO . putStrLn . formatBlock args =<< getSSID
 
-format' :: Text -> Text
-format' ssid' = "\61931" <> " " <> ssid'
+formatBlock :: [String] -> Text -> Text
+formatBlock args ssid
+  = if Text.null ssid
+  then fromMaybe "No WiFi" $ getNoWifiIcon args
+  else (fromMaybe "" $ getWifiIcon args) <> ssid
 
-ssid :: Shell Text
-ssid = strict $ inshell "iwgetid -r" empty
+getNoWifiIcon :: [String] -> Maybe Text
+getNoWifiIcon icons = case icons of
+  noWifiIcon : _ -> Just $ Text.pack noWifiIcon
+  _              -> Nothing
+
+getWifiIcon :: [String] -> Maybe Text
+getWifiIcon icons = case icons of
+  _ : wifiIcon : _ -> Just $ Text.pack wifiIcon
+  _                -> Nothing
+
+getSSID :: Shell Text
+getSSID = do
+  exitCode <- shell "iwgetid" empty
```

Comment:
Somehow my i3bar is displaying the output of `iwgetid` instead of `iwgetid -r`

```
$ iwgetid 
wlp3s0    ESSID:"vtec_ch_9000rpm"
$ iwgetid -r
vtec_ch_9000rpm
```
![shot1586354401](https://user-images.githubusercontent.com/837104/78793064-cfbe2380-79a9-11ea-9aa6-4cffa06cf19b.png)